### ISSUE TITLE
AppInfo class for unpackaged apps

### ISSUE THREAD

**Issue by jaigak** [microsoft/WindowsAppSDK#1083]
Today the [AppInfo](https://docs.microsoft.com/en-us/uwp/api/Windows.ApplicationModel.AppInfo?view=winrt-20348) provides information about an app. However this only works for packaged apps. Please update the class to support unpackaged apps as well. For unpackaged apps, it could pull the information from the file details instead of the manifest as unpackaged apps don't have a manifest.

**Comment by Jaiganeshkumaran** [comment]
Windows Server 2022 has added a SupportedFileExtensions property which gives the list of registered file types in the manifest. This can be useful so the apps can declare the file types once and use them for other things like file picker however since most developers need to target older versions, they can't use it. It should be possible to reimplement such functionality by reading the manifest, so could be a separate lifted Reunion API that also supports unpackaged apps.

**Comment by riverar** [comment]
Populating appmodel's AppInfo object with unmodeled app info seems like the wrong approach.

Retrieving icons and other metadata would be very helpful for app pickers, our app (audio mixer), etc. Perhaps a separate undocked API proposal is in order.

**Comment by DrusTheAxe** [comment]
+1

[Windows.ApplicationModel.AppInfo](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.AppInfo) has properties and methods that are meaningless and problematic for unpackaged apps. In fact most of AppInfo's properties and methods are hard to imagine how to do something useful for unpackaged apps -- Package, PackageFamilyName, and more. Not to mention the very basic basic question how do you identify an "app"? Microsoft.ApplicationModel.AppInfo.Current could use current process as the context, but what do you use for AppInfo.GetFromAppUserModelId() or other lookups?

Are you really looking for AppInfo or [AppDisplayInfo](https://docs.microsoft.com/uwp/api/windows.applicationmodel.appdisplayinfo)? The latter provides a DisplayName, Description and Logo, for display purposes (e.g. localized for human consumption).

What information are you looking for? Please list the properties of interest (be it in AppInfo, AppDisplayInfo, other APIs or none at all)

What are you looking to do with the data? Can you outline some scenarios to help me (and others) understand what you're looking to accomplish? Much appreciated.

**Comment by Jaiganeshkumaran** [comment]
> +1
> 
> [Windows.ApplicationModel.AppInfo](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.AppInfo) has properties and methods that are meaningless and problematic for unpackaged apps. In fact most of AppInfo's properties and methods are hard to imagine how to do something useful for unpackaged apps -- Package, PackageFamilyName, and more. Not to mention the very basic basic question how do you identify an "app"? Microsoft.ApplicationModel.AppInfo.Current could use current process as the context, but what do you use for AppInfo.GetFromAppUserModelId() or other lookups?
> 
> Are you really looking for AppInfo or [AppDisplayInfo](https://docs.microsoft.com/uwp/api/windows.applicationmodel.appdisplayinfo)? The latter provides a DisplayName, Description and Logo, for display purposes (e.g. localized for human consumption).
> 
> What information are you looking for? Please list the properties of interest (be it in AppInfo, AppDisplayInfo, other APIs or none at all)
> 
> What are you looking to do with the data? Can you outline some scenarios to help me (and others) understand what you're looking to accomplish? Much appreciated.

Right, AppDisplayInfo seems a better choice however AppDisplayInfo is obtained through AppInfo.DisplayInfo.

**Comment by Jaiganeshkumaran** [comment]
Here's what I think the API should look like.
```
[default_interface]
runtimeclass AppInfo
{
    String Name { get; };
    String Description { get; };
    String AppUserModelId { get; };
    Windows.ApplicationModel.PackageVersion Version { get; };
}
```
For packaged apps, the class will simply pull in the information from the system AppInfo class whereas, in unpackaged apps, information will be retrieved from the executable (set in RC file). Information that's part of the package but also makes sense for unpackaged apps like the app version should also be accessible through this class.

**Comment by DrusTheAxe** [comment]
> the system AppInfo class

What is that?

The only `AppInfo` class I'm aware of is [Windows.ApplicationModel.AppInfo](https://learn.microsoft.com/uwp/api/windows.applicationmodel.appinfo) discussed above and inappropos for unpackaged apps.

You're suggesting a _new_ object containing this info?

Where's the data come from?
* Packaged -- Windows.ApplicationModel.AppInfo (Name, Publisher, Version, Description, Logo, AppUserModelId)
* Unpackaged -- EXE's VERSIONINFO (Name,Publisher,Version,Description?,Logo), and no AppUserModelId?

How does one ask for this info?
* Current process?
* Specific process?
* File (exe) on disk?
* Path (directory) on disk?

Current process is easy to grok for packaged+unpackaged

How do you specify a target process to retrieve this info? How does Windows reason over the target to retrieve this info?

File/Path (inert/not-process) targets are easy for EXE if the data's just info from an EXE's VERSIONINFO but inherently difficult for a package (path->package xref is exceedingly difficult, fragile and non-performant, hence the reason you don't see any such APIs...)

I get the well intended desire but I don't how it's tractable or actionable.

Do you?
